### PR TITLE
Workaround PlatformNotSupportedException on UWP

### PR DESCRIPTION
**Pull request by natemcmaster**
Add a workaround for a corner case in .NET Native where TypeInfo.IsAssignableFrom throws `PlatformNotSupportedException`.

@MichaelKetting This is just the first of what could be several changes. I'll update this branch in my work as I find more fixes that are necessary. But I'm submitting this one early because it affects all usages of `QueryParser`.

### CODE REVIEW

File: Core/Parsing/Structure/IntermediateModel/SupportedMethodSpecifications.cs [hunk 1]
```diff
@@ -96,12 +96,27 @@ private static bool HasIndexSelectorParameter (MethodInfo methodInfo, int parame
...
     }
+
+    private static TypeInfo UnwrapEnumerable (this TypeInfo typeInfo)
+    {
+      var comparedTypeInfo = typeInfo;
+
+#if !NET_4_0 && !NET_3_5
+      // Workaround for a corner case in .Net Native because
+      // IsAssignableFrom throws for open generics.
+      if (typeInfo.ContainsGenericParameters)
```
Comment (MichaelKetting) [inline Core/Parsing/Structure/IntermediateModel/SupportedMethodSpecifications.cs]: That's a great idea! Totally obvious solution once you see it in code :)

**Comment by MichaelKetting** [comment]
@natemcmaster Great! Will be monitoring this PR for further updates and questions :+1:

File: Core/Parsing/Structure/IntermediateModel/SupportedMethodSpecifications.cs [hunk 2]
```diff
@@ -96,12 +96,27 @@ private static bool HasIndexSelectorParameter (MethodInfo methodInfo, int parame
...
+        comparedTypeInfo = typeInfo.BaseType.GetTypeInfo();
```
Comment (divega) [inline Core/Parsing/Structure/IntermediateModel/SupportedMethodSpecifications.cs]: Would we ever need to do this more than once until we find a base type that isn't generic?

Comment (MichaelKetting) [inline Core/Parsing/Structure/IntermediateModel/SupportedMethodSpecifications.cs]: Good catch! Thought about this, too, when I first saw it but given that it's only used in a very limited context (Enumerable and Queryable) and the type hierarchy employed by the parameter types of Enumerable and Queryable, one hop is enough. Plus, we'll also have unit tests covering 100% of the method lookups, so we're safe against missed cases, too.

Now that I re-think this, the only mishap that could happen, were if a new overload would be introduced for the relevant operations and that overload took a parameter at the specific position with more than one open generic hop in it's type hierharchy. Before now, I considered this to be highly unlikely, based on past development of Enumerable and Queryable, but I'm also okay with a loop to find the first non-generic base type (or null). That way, we'd be 100% bullet proof to future changes at this corner of the system.

**Pull request merged**